app/src/main/java/woowacourse/shopping/feature/order/order/OrderPresenter.kt

```diff
@@ -0,0 +1,98 @@
+package woowacourse.shopping.feature.order.order
+
+import com.example.domain.repository.CartRepository
+import com.example.domain.repository.OrderRepository
+import woowacourse.shopping.feature.order.Discount
+import woowacourse.shopping.mapper.toPresentation
+import woowacourse.shopping.model.CartProductUiModel
+
+class OrderPresenter(
+    private val cartIds: List<Long>,
+    private val view: OrderContract.View,
+    private val cartRepository: CartRepository,
+    private val orderRepository: OrderRepository,
+) :
+    OrderContract.Presenter {
+    private var orderProducts: List<CartProductUiModel> = listOf()
+    private var discountCondition: Discount.Condition? = null
+    private var totalPrice = 0
+    private var payAmount = 0
+
+    override fun requestProducts() {
+        cartRepository.getAll(
+            onSuccess = { it ->
+                val products = mutableListOf<CartProductUiModel>()
+                val cartProducts = it.map { it.toPresentation() }
+                cartIds.forEach { cartId ->
+                    val cartProduct: CartProductUiModel? = cartProducts.find { it.cartId == cartId }
+                    if (cartProduct != null) products.add(cartProduct)
```

Comment:
```suggestion
                    cartProduct?.let { products.add(it) }
```
코틀린스럽게 이렇게 바꿔봐도 되겠네요.